### DISCUSSION TITLE
Chain not imported

### DISCUSSION THREAD

**Discussion by johnpaulcas** [wevm/wagmi#2579] [Q&A]
We are having an issue when the network is not yet added on metamask, what's the best practice to handle this kind of error?

```
MetaMask - RPC Error: Unrecognized chain ID "0x13881". Try adding the chain using wallet_addEthereumChain first.
```

EDIT:
also this cause an issue when doing the transaction after import
```
MetaMask - RPC Error: [ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":-32000,"message":"header not found"}}}
```

**Comment by SaadNiaziDev** [comment]
can u share code snippet like how are u doing this?
